Translation support

aliseyalvi opened the pull request:
### What does this PR?
This PR adds support for content translation in waves screen

### Steps to reproduce
Go to Waves screen -> open post options from 3 dot menu -> tap on translate option
It would open a translation sheet with different languages selector and translated content

### Issue number
fixes #2805 

### Screenshots/Video

https://github.com/ecency/ecency-mobile/assets/48380998/f4c8b7ab-4d10-4f2a-a685-3c15d3f0cb5d

noumantahir reviewed (changes requested):
- add better support for light and dark themes
- modal goes too up high, either make it dynamic based on content length, or make sure it does not go behind notch
- may be rearrange post options items so translate item it little more visible

<img width="197" alt="Screenshot 2024-01-16 at 14 48 53" src="https://github.com/ecency/ecency-mobile/assets/6298342/1812b305-68e7-4b0f-a951-78fe5a68cc4b">
<img width="197" alt="Screenshot 2024-01-16 at 14 48 09" src="https://github.com/ecency/ecency-mobile/assets/6298342/30230ae1-c174-43fb-a232-4e336057c59a">

aliseyalvi commented:
<img width="342" alt="image" src="https://github.com/ecency/ecency-mobile/assets/48380998/eb431ac4-c06b-43e4-a7d3-b411389e8149">
<img width="342" alt="image" src="https://github.com/ecency/ecency-mobile/assets/48380998/440f18ca-f84a-4b4b-95b6-dbbf42eb77cd">

noumantahir commented on src/components/post-translation-modal/postTranslationModal.tsx:
```diff
@@ -0,0 +1,189 @@
+  const _renderLanguageSelector = () => (
+    <View style={styles.languageSelectorRow}>
+      <View style={styles.row}>
+        <Text style={styles.labelText}>From</Text>
```
use intl

noumantahir commented on src/components/post-translation-modal/postTranslationModal.tsx:
```diff
@@ -0,0 +1,189 @@
+        />
+      </View>
+      <View style={styles.row}>
+        <Text style={[styles.labelText, styles.toText]}>To</Text>
```
use intl

noumantahir commented on src/components/post-translation-modal/postTranslationModal.tsx:
```diff
@@ -0,0 +1,189 @@
+        <Text style={[styles.labelText, styles.toText]}>To</Text>
+        <SelectDropdown
+          data={supportedLangsList.filter(
+            (item) => item.code !== srcLang.code || item.code !== selectedSourceLang.code,
```
possible to default `To` language to selected app language?

noumantahir reviewed (changes requested):
overall good, two tiny change and one conditional change requested

aliseyalvi commented:
@noumantahir 

1. added intl
2. added support for auto select target lang according to app default language

https://github.com/ecency/ecency-mobile/assets/48380998/b18a05c9-626a-48c0-8479-122ff6cd93c9

noumantahir reviewed (approved):
functionality wise it's smooth and very useable, user experience can get better as we iterate later.
